engine/gamesys/src/gamesys/test/test_gamesys.cpp

```diff
@@ -834,6 +835,79 @@ TEST_F(BufferMetadataTest, MetadataLuaApi)
     dmGameSystem::FinalizeScriptLibs(scriptlibcontext);
 }
 
+struct ScriptComponentTestData
+{
+    dmGameObject::HCollection m_Collection;
+    const char*               m_ComponentType;
+};
+
+static int ScriptComponentTestCallback(lua_State* L)
+{
+    lua_getglobal(L, "test_data");
+    ScriptComponentTestData* data = (ScriptComponentTestData*)lua_touserdata(L, -1);
+    lua_pop(L, 1);
+
+    dmGameObject::HComponent out_component = 0;
+    dmGameObject::GetComponentFromLua(L, 1, data->m_Collection, data->m_ComponentType, &out_component, 0, 0);
+
+    // We should have an actual pointer at this stage, and it is not likely it is less than a certain low number
+    lua_pushboolean(L, (uintptr_t)out_component > 100000);
```

Comment:
The test is simple.
We don't check for validity per se, but rather if it looks to be an index or not.